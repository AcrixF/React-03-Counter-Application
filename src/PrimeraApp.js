//Functional Components
import React from 'react';
import PropTypes from 'prop-types';

const PrimeraApp = ({greet, subtitle}) => {
 
    return (
        <>
            <h1>{ greet }</h1>
            {/* <pre>{JSON.stringify(user, null, 3)}</pre> */}
            <p>{ subtitle }</p>
        </>
    );
};

PrimeraApp.propTypes = {
    greet: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
    subtitle : 'Learning React!'
}

export default PrimeraApp;

