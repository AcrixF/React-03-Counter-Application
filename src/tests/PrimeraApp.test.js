import '@testing-library/jest-dom';
import React from 'react';
//import { render } from '@testing-library/react';
import PrimeraApp from '../PrimeraApp';

import { shallow } from 'enzyme';


describe('Test Component <PrimeraApp/>', () => {

    // test("Should show the message" , () => {
    //     const greet = "Brenda Marlen";
    //     const { getByText } = render(<PrimeraApp greet={greet}/>);
    //     expect(getByText(greet)).toBeInTheDocument();
    // });

    test("Should show the <PrimeraApp/> correctly", () => {
        const greet = "Brenda Marlen";

        const wrapper = shallow(<PrimeraApp greet={ greet }/>);

        expect(wrapper).toMatchSnapshot();
    });

    test('should show the subtitle message correctly', () => {
        const greet = "Brenda Marlen";
        const subtitle = 'Learning React!';
        const wrapper = shallow(<PrimeraApp greet= { greet } subtitle= {subtitle}/>); 
        const validate = wrapper.find('p').text();

        expect(subtitle).toBe(validate);

    });
    

});