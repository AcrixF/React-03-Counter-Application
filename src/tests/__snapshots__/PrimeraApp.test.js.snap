// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Component <PrimeraApp/> Should show the <PrimeraApp/> correctly 1`] = `
<Fragment>
  <h1>
    Brenda Marlen
  </h1>
  <p>
    Learning React!
  </p>
</Fragment>
`;
